index.js

```diff
@@ -1,14 +1,200 @@
+          ts
+        };
+      },
+      // A required interface for the asset module.
+      async startDevServer(options) {
+        self.buildOptions = options;
+        self.shouldCreateDevServer = true;
+        await self.buildBefore(options);
+
+        const {
+          scenes: currentScenes,
+          build: currentBuild
+        } = self.getCurrentMode(options.devServer);
+
+        self.entrypointsManifest.unshift(self.getViteClientEntrypoint(currentScenes));
+
+        let ts;
+        if (currentBuild === 'public') {
+          await self.buildPublic(options);
+        }
+        if (currentBuild === 'apos') {
+          ts = await self.buildApos(options);
+        }
+
+        const viteManifest = await self.getViteBuildManifest(currentBuild);
+        self.entrypointsManifest = await self.applyManifest(
+          self.entrypointsManifest, viteManifest
+        );
+
+        return {
+          entrypoints: self.entrypointsManifest,
+          hmrTypes: [ ...new Set(
+            self.getBuildEntrypointsFor(options.devServer)
+              .map((entry) => entry.type)
+          ) ],
+          ts,
+          devServerUrl: self.getDevServerUrl()
+        };
+      },
+      // A required interface for the asset module.
+      // Initialize the watcher for triggering vite HMR via file
+      // copy to the build source.
+      // `chokidar` is a chockidar `FSWatcher` or compatible instance.
+      async watch(chokidar, buildOptions) {
+        self.buildWatchIndex();
+        // Initialize our voting system to detect what entrypoints
+        // are concerned with a given source file change.
+        self.setWatchVoters(
+          self.getBuildEntrypointsFor(buildOptions.devServer)
+        );
+
+        chokidar
+          .on('add', (p) => self.onSourceAdd(p, false))
+          .on('addDir', (p) => self.onSourceAdd(p, true))
+          .on('change', (p) => self.onSourceChange(p))
+          .on('unlink', (p) => self.onSourceUnlink(p, false))
+          .on('unlinkDir', (p) => self.onSourceUnlink(p, true));
+      },
+      // A required interface for the asset module.
+      // This method is called when build and watch are not triggered.
+      // Enhance and return any entrypoints that are included in the manifest
+      // when an actual build/devServer is triggered.
+      // The options are same as the ones provided in the `build` and
+      // `startDevServer` methods.
+      async entrypoints(options) {
+        const entrypoints = self.apos.asset.getBuildEntrypoints(options.types)
+          .filter(entrypoint => entrypoint.condition !== 'nomodule');
+
+        if (options.devServer) {
+          const { scenes } = self.getCurrentMode(options.devServer);
+          entrypoints.unshift(self.getViteClientEntrypoint(scenes));
+        }
+
+        return entrypoints;
+      },
+      // A required interface for the asset module.
+      async clearCache() {
+        await fs.remove(self.cacheDirBase);
```

Comment:
If we use `node:fs/promises` instead

```suggestion
        await fs.rmdir(self.cacheDirBase, { recursive: true });
```

Reply:
Same as above